src/Swift.Runtime/src/Swift/SwiftEquatable.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices.Swift;
+using Swift.Runtime.InteropServices;
+
+namespace Swift.Runtime
+{
+    public interface ISwiftEquatable
+    {
+        // This interface is intentionally left empty. It serves as a marker for types that implement Swift's Equatable protocol.
+    }
+
+    /// <summary>
+    /// Provides functionality to use Swift's Equatable protocol for equality comparison.
+    /// </summary>
+    public static class SwiftEquatable
+    {
+        // P/Invoke declaration for Swift's Equatable protocol equality operator
+        [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+        [DllImport(KnownLibraries.SwiftCore, EntryPoint = "$sSQ2eeoiySbx_xtFZTj")]
+        private static extern bool PInvoke_SwiftEquals(
+            IntPtr lhs,
+            IntPtr rhs,
+            SwiftSelf typeMetadataInSwiftSelf,
+            TypeMetadata typeMetadata,
+            ProtocolWitnessTable equatableProtocolWitnessTable);
+
+        /// <summary>
+        /// Compares two objects using Swift's Equatable protocol.
+        /// </summary>
+        /// <typeparam name="T">Type that implements IEquatable&lt;T&gt;</typeparam>
+        /// <param name="lhs">Left-hand side object</param>
+        /// <param name="rhs">Right-hand side object</param>
+        /// <returns>True if the objects are equal according to Swift's equality</returns>
+        public static unsafe bool Equals<T>(T lhs, T rhs)
```

Comment:
Should we add a generic constrain that `T` must implment `ISwiftEquatable`?
We requrie the `T` to provide PWT to `ISwiftEquatable` so it's close to that already anyway.

Reply:
I do not think we emit interface conformance as of now. That is a type conforming to SwiftEquitable will not have `: ISwiftEquitable` on its definition, making it fail the compile time check. 

Since `Swift.Int` implements `Equatable` this brings us back to the boxing discussion